src/v4/core/reaction.ts

```diff
@@ -224,6 +227,22 @@ function runReactionsHelper() {
     const allReactions = globalState.pendingReactions
     let iterations = 0
 
+    // Calculate how many times the reaction helper is triggered
+    //  inside a single microtask, and warn if it is too many.
+    if (allReactions.length > 0) {
+        if (curMicrotaskReactionCount === 0) {
+            Promise.resolve().then(() => {
+                if (curMicrotaskReactionCount > REACTIONS_IN_MICROTASK_LOOP_WARN_THRESHOLD) {
+                    console.warn(
+                        `MOBX WARNING, a single microtask loop had ${curMicrotaskReactionCount} reactions (which are caused by changes to observables). Batch these actions with @action or runInAction to gain up to a ${curMicrotaskReactionCount}X performance boost!`
+                    )
+                }
+                curMicrotaskReactionCount = 0
+            })
+        }
+        curMicrotaskReactionCount++
```

Comment:
My personal nit, I like to see that number explicitly there.
```suggestion
        curMicrotaskReactionCount += 1
```